### PR TITLE
docs: squad retrospective — update agent definitions after PR #331

### PR DESCRIPTION
**Pull request by daviburg**
## Summary

Agent definition updates from first squad retrospective (PR #331 heartbeat wrapper).

Each agent reflected on what went wrong and proposed self-improvements. The coordinator synthesized and dispatched this PR.

### Coordinator
- Enforce full squad loop (develop → test → review → iterate)
- Check automated PR review comments
- Never do specialist work

### Developer
- Smoke test new scripts beyond pytest
- Address automated review comments before marking PR done
- Platform-specific considerations section

### Tester
- Code review before testing
- Always test failure paths
- Platform gotchas checklist

### Reviewer
- Exit code propagation as blocking-level concern
- Verify automated review comments addressed
- Verify PR conversations resolved
- Severity calibration guidance

Closes no issues — this is a process improvement PR.

### CODE REVIEW

File: .github/agents/tester.agent.md [hunk 1]
```diff
@@ -12,25 +12,41 @@ You are the QA and testing specialist for narrative-state-engine. Your job is to
...
-3. Compare results against expected ground truth in `tests/fixtures/`
-4. Report failures with: test name, expected vs actual, likely root cause
-5. Write new tests when gaps in coverage are identified
+1. **Code review**: Read the code under test and review for correctness issues, platform gotchas, and security concerns before running tests.
+2. Run `pytest` to get baseline test status
+3. For extraction validation: run `tools/validate_extraction.py` with appropriate flags
+4. Compare results against expected ground truth in `tests/fixtures/`
+5. Report failures with: test name, expected vs actual, likely root cause
+6. Write new tests when gaps in coverage are identified
+- For CLI tools and wrappers: explicitly test exit code propagation (success=0, failure=non-zero, signal/interrupt), stderr vs stdout separation, and behavior under bad input or missing dependencies.
```
[the remark this answers is not in the document]
Comment (daviburg) [reply .github/agents/tester.agent.md]: Fixed in c9fbeb5 ? integrated the CLI wrapper testing bullet as numbered step 7 in the approach list, restoring the numbered sequence.

File: .github/agents/developer.agent.md [hunk 2]
```diff
@@ -26,10 +26,12 @@ You are the code developer for narrative-state-engine. Your job is to implement
...
-6. **Commit**: Use conventional commit prefixes (`fix:`, `feat:`, `docs:`, `chore:`).
-7. **PR**: Create with `gh pr create --body-file` — never inline `--body`.
+4. **Test**: Run the full test suite. Add tests for new functionality. For scripts and CLI tools that can't be unit-tested, run manual smoke tests (success case, failure case, edge cases) and document the commands in the PR body.
+5. **Smoke test**: For new scripts, wrappers, or CLI tools, execute them on the target platform(s) beyond pytest. Verify they launch, produce expected output, handle errors, and exit cleanly.
+6. **Document**: Update architecture.md, roadmap.md, or usage.md as needed (Rule 8).
+7. **Commit**: Use conventional commit prefixes (`fix:`, `feat:`, `docs:`, `chore:`).
+8. **PR**: Create with `gh pr create --body-file` — never inline `--body`.
+9. **Review feedback**: After creating a PR, check for automated review comments (Copilot, CodeQL, linters). Address each comment with a code fix or explanation. Respond to each comment on the PR. Do not consider the PR complete until all automated review comments are resolved.
```
[the remark this answers is not in the document]
Comment (daviburg) [reply .github/agents/developer.agent.md]: Fixed in c9fbeb5 ? consolidated steps 4 (Test) and 5 (Smoke test) into a single step 4 that covers both pytest and manual smoke testing. Remaining steps renumbered (8 steps total, down from 9).
